src/ServiceControl.Audit/Auditing/AuditPersister.cs

```diff
@@ -157,8 +157,7 @@ await bodyStorageEnricher.StoreAuditMessageBody(context.Body, context.Headers, m
 
                 var auditMessage = new ProcessedMessage(context.Headers, new Dictionary<string, object>(metadata))
                 {
-                    // We do this so Raven does not spend time assigning a hilo key
-                    Id = $"ProcessedMessages/{Guid.NewGuid()}"
+                    Id = $"ProcessedMessages/{context.Headers.UniqueId()}"
```

Comment:
The `UniqueId()` method can fail if headers are missing. I think we should use here similar method but fall back to `Guid.NewGuid()` if the header-based deterministic ID generation fails.

Reply:
Can you have a look now?